experiments/kompanion/cmd/export/options.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package export
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/spf13/pflag"
+)
+
+const (
+	// flag names.
+	kubeconfigFlag       = "kubeconfig"
+	reportNamePrefixFlag = "report-prefix"
+
+	targetNamespacesFlag = "target-namespaces"
+	ignoreNamespacesFlag = "exclude-namespaces"
+
+	targetObjectsFlag = "target-objects"
+	ignoreObjectsFlag = "exclude-objects"
+
+	workerRoutinesFlag = "worker-routines"
+)
+
+type ExportOptions struct {
+	kubeconfig       string
+	reportNamePrefix string
+
+	targetNamespaces []string
+	ignoreNamespaces []string
+
+	targetObjects []string
+	ignoreObjects []string
+
+	workerRoutines int
+}
+
+func (o *ExportOptions) AddFlags(flags *pflag.FlagSet) {
+	flags.StringVarP(&o.kubeconfig, kubeconfigFlag, "", o.kubeconfig, "path to the kubeconfig file.")
+	flags.StringVarP(&o.reportNamePrefix, reportNamePrefixFlag, "", o.reportNamePrefix, "Prefix for the report name. The tool appends a timestamp to this in the format \"YYYYMMDD-HHMMSS.milliseconds\".")
+
+	flags.StringArrayVarP(&o.targetNamespaces, targetNamespacesFlag, "", o.targetNamespaces, "namespace prefix to target the export tool. Targets all if empty. Can be specified multiple times.")
+	flags.StringArrayVarP(&o.ignoreNamespaces, ignoreNamespacesFlag, "", o.ignoreNamespaces, "namespace prefix to ignore. Excludes nothing if empty. Can be specified multiple times. Defaults to \"kube\".")
+
+	flags.StringArrayVarP(&o.targetObjects, targetObjectsFlag, "", o.targetObjects, "object name prefix to target. Targets all if empty. Can be specified multiple times.")
+	flags.StringArrayVarP(&o.ignoreObjects, ignoreObjectsFlag, "", o.ignoreObjects, "object name prefix to ignore. Excludes nothing if empty. Can be specified multiple times.")
+
+	flags.IntVarP(&o.workerRoutines, workerRoutinesFlag, "", o.workerRoutines, "Configure the number of worker routines to export namespaces with. Defaults to 10. ")
+}
+
+func (opts *ExportOptions) validateFlags() error {
+	if opts.workerRoutines <= 0 || opts.workerRoutines > 100 {
+		return fmt.Errorf("invalid value %d for flag %s. Supported values are [1,100]", opts.workerRoutines, workerRoutinesFlag)
+	}
+
+	return nil
+}
+
+func (o *ExportOptions) Print() {
+	log.Printf("kubeconfig set to %q.\n", o.kubeconfig)
```

Comment:
Nit: I feel like we might as well print the JSON format or %+v.  It's going to be easy to forget to print a field here.

Reply:
If this were the actual kubeconfig that would make sense. I believe this is just the path to a kube config file.